ChronoKeeper/ChronoKeeperInstance.cpp

```diff
@@ -0,0 +1,99 @@
+
+#include <arpa/inet.h>
+
+//#include "Storyteller.h"
+
+
+#include "KeeperIdCard.h"
+#include "KeeperStatsMsg.h"
+#include "KeeperRecordingService.h"
+#include "KeeperRegClient.h"
+#include "LogIngestionQueue.h"
+
+#define KEEPER_GROUP_ID 7
+#define KEEPER_REGISTRY_SERVICE_NA_STRING  "ofi+sockets://127.0.0.1:1234"
+#define KEEPER_REGISTRY_SERVICE_PROVIDER_ID	25
+
+#define KEEPER_RECORDING_SERVICE_PROTOCOL  "ofi+sockets"
+#define KEEPER_RECORDING_SERVICE_IP  "127.0.0.1"
+#define KEEPER_RECORDING_SERVICE_PORT  "5555"
+#define KEEPER_RECORDING_SERVICE_PROVIDER_ID	22
+#define IP_ADDRESS "127.0.0.1"
+
+int main(int argc, char** argv) {
+
+  int exit_code = 0;
```

Comment:
Where is the integration with the ChronoVisor? Aren't the keepers supposed to send their ip address, port number and machine configuration to the ChronoVisor and the Visor fills/updates a KeeperTable? 
The keepers are also supposed to verify clients by checking if a client id exists in the ChronoVisor ClientRegistry.

Reply:
take a look at KeeperRegistry, KeeperRegistrationService  that would be in the ChronoVisor process .
ChronoKeeperInstance only contains KeeperRegistryClient side of the communication